### PR TITLE
Show warning when module scripts are ignored

### LINKED ISSUE #24089: Warn when module scripts are ignored
This is an ongoing source of confusion when pages don't work that depend on module scripts. We should log a warning when they are encountered as long as we don't support them.

### PR DESCRIPTION
**Pull request by CYBAI**
The first five commits are `cherry-pick`-ed from the module script PR.
I think it might be also good to have this PR first so that reviewers can focus on module script things on that PR!

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #24089
- [x] These changes do not require tests because it just ignored module scripts (or should we turn on module script tests in this PR and update those test expectation to TIMEOUT?)

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/24142)

### CODE REVIEW

**Comment by jdm** [comment]
@bors-servo r+
Thanks!

File: components/script/dom/htmlscriptelement.rs [hunk 1]
```diff
@@ -695,45 +702,62 @@ impl HTMLScriptElement {
...
-        let is_js = match type_attr.as_ref().map(|s| s.value()) {
-            Some(ref s) if s.is_empty() => {
-                // type attr exists, but empty means js
+        let language_attr = element.get_attribute(&ns!(), &local_name!("language"));
+
+        let script_type = match (
+            type_attr.as_ref().map(|t| t.value()),
+            language_attr.as_ref().map(|l| l.value()),
+        ) {
+            (Some(ref ty), _) if ty.is_empty() => {
```
Comment (jdm) [inline components/script/dom/htmlscriptelement.rs]: I think this branch is causing the new test failures.

Comment (jdm) [reply components/script/dom/htmlscriptelement.rs]: Nevermind, that is incorrect.

File: components/script/dom/htmlscriptelement.rs [hunk 2]
```diff
@@ -695,45 +702,62 @@ impl HTMLScriptElement {
...
-                // https://github.com/rust-lang/rust/issues/21114
-                is_js
+            (None, Some(ref lang)) => {
+                debug!("script language={}", &***lang);
+                let language = format!("text/{}", &***lang);
+
+                if SCRIPT_JS_MIMES.contains(
+                    &language
+                        .to_ascii_lowercase()
+                        .trim_matches(HTML_SPACE_CHARACTERS),
```
Comment (jdm) [inline components/script/dom/htmlscriptelement.rs]: Looks like we did not used to call trim_matches, and the tests that are failing have trailing space characters.

Comment (CYBAI) [reply components/script/dom/htmlscriptelement.rs]: Oh, yeah! I think I mixed this step up with the `has type attribute` one! Let me fix it! Thanks!

**Pull request merged**